src/flake8_numbers/check_numbers.py

```diff
@@ -0,0 +1,209 @@
+"""A flake8 plugin to check for numbers and their readability."""
+
+import ast
+from dataclasses import dataclass
+from importlib import metadata
+from typing import Any, Iterable, Optional, Tuple, Type
+
+
+@dataclass
+class ErrorReport:
+    """A class to represent an error report."""
+
+    line: int
+    """The line number of the error."""
+
+    column: int
+    """The column number of the error."""
+
+    message: str
+    """The error message."""
+
+
+def _base_value(number_literal: str) -> int:
+    """Get the base value of the given number literal.
+
+    Args:
+        number_literal: The number literal to get the base value of.
+
+    Returns:
+        The base value of the given number literal.
+    """
+    lower_number_literal = number_literal.lower()
+    if lower_number_literal.startswith("0b"):
+        return 2
+    if lower_number_literal.startswith("0o"):
+        return 8
+    if lower_number_literal.startswith("0x"):
+        return 16
+    return 10
+
+
+def _separator_modulo_for_base(base: int) -> int:
+    """Get the separator modulo for the given base.
+
+    Note: We keep this currently as helper function, to make that part of the code
+    more readable as well as open up the possibility of making this configurable
+    in the future.
+
+    Args:
+        base: The base to get the separator modulo for.
+
+    Returns:
+        The separator modulo for the given base.
+    """
+    match base:
+        case 2:
+            return 4
+        case 8:
+            return 4
+        case 16:
+            return 4
+        case _:
+            return 3
```

Comment:
Just FYI:

```suggestion
        case 2 | 8 | 16:
            return 4
        case _:
            return 3
```

Reply:
Thanks. I've left it spanned out for the reason of future configurability of these values. Then using `case 2 | 8 | 16` would be needed to be undone again. :)